apps/examples/kernel_sample/kernel_sample_main.c

```diff
@@ -300,11 +243,11 @@ static int user_main(int argc, char *argv[])
 
 	unsetenv(g_var1_name);
 	show_environment(false, true, true);
-	check_test_memory_usage();
+	check_test_memory_usage(&g_mmprevious, &g_mmafter);
```

Comment:
I think we don't need to keep the global variables here. Let's move them into new file and keep original usages.